SPECS/golang/golang.spec

```diff
@@ -68,7 +71,7 @@ popd
 rm -rf %{_libdir}/golang
 
 # Make go%{bootstrap_compiler_version} as the new bootstrapper
-mv -v %{_topdir}/BUILD/go1.19.12 %{_libdir}/golang
+mv -v %{_topdir}/BUILD/go%{bootstrap_compiler_version} %{_libdir}/golang
 
 # Build current go version
 export GOHOSTOS=linux
```

Comment:
Can you verify that these exports are correct?  Esepcially the "GOROOT" one.  I have a bug against Mariner 2.0 (https://github.com/microsoft/CBL-Mariner/issues/6363) that suggests GOROOT is pointing to the wrong place. 

     It seems to be setting GOROOT to /usr/bin/go.
     But /usr/bin/go is a binary.
     Actual GOROOT should be /usr/lib/golang in this case.

Reply:
The GOROOT_FINAL support has been dropped in go1.21.x
https://github.com/golang/go/issues/62047

Reply:
Yeah, that's the wrong place. It should be possible to simply remove it--since 1.9, GOROOT is automatic. Although there might be something odd necessary due to bootstrapping, I don't have practical experience with it.

Also, the only part of `/etc/profile.d/go-exports.sh` that I think makes a difference nowadays is `GOPATH`, but I don't know whether it's actually good or not to move it from a user-specific path (default) to a systemwide path. For a container I suppose it doesn't matter.

I tried `go env` before and after `. /etc/profile.d/go-exports.sh`, and the difference was:
```diff
-GOMODCACHE="/root/go/pkg/mod"
+GOMODCACHE="/usr/share/gocode/pkg/mod"
-GOPATH="/root/go"
+GOPATH="/usr/share/gocode"
```

(I don't know in what circumstances a bad `GOROOT_FINAL` results in an error, this isn't something I've seen before.)

I wasn't able to get https://github.com/microsoft/CBL-MarinerTutorials/tree/main/build-in-container to work, so I don't have a good dev loop to try things out. (We decided to go ahead with this PR because if URL substitution fails, it probably won't be a subtle problem. 😄) I might have been hitting the same issues as our buddy builds have been hitting. This is something I'll have to look into later.